File: idea-plugin/src/main/java/com/lingce/cleaner/FileClassification.java
package com.lingce.cleaner;

final class FileClassification {
    enum Category {
        SAFE,
        TEMPORARY,
        CONFIGURATION,
        DISPOSABLE_AI,
        SUSPICIOUS_AI,
        UNKNOWN
    }

    private final Category category;
    private final String reason;

    FileClassification(Category category, String reason) {
        this.category = category;
        this.reason = reason;
    }

    Category category() {
        return category;
    }

    String reason() {
        return reason;
    }
}

File: idea-ai-cleaner/src/main/kotlin/com/lingce/cleaner/model/FileClassification.kt
package com.lingce.cleaner.model

enum class FileCategory {
    SAFE,
    TMP,
    AI_GENERATED,
    PROJECT_CONFIG,
    AI_CONFIG,
    SUSPICIOUS
}

data class FileClassification(
    val category: FileCategory,
    val reason: String,
    val confidence: Double = 1.0
)
